include/fluent-bit/flb_aws_util.h

```diff
@@ -177,5 +177,17 @@ int flb_read_file(const char *path, char **out_buf, size_t *out_size);
 flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
                          char *tag_delimiter, uint64_t seq_index);
 
+/*
+ * Count the substring number in string.
+ */
+int count_substring(char *str1, char *str2);
+
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
+int create_time_format_string(char **out_buff, size_t *out_size,
+                              char *time_key_format, const struct tm *timeptr,
+                              size_t len, struct flb_time *tms);
```

Comment:
Should this function take both a `struct tm *timeptr` as arg and also a `struct flb_time *tms`?

Both are time values, and both are the same time right? the caller code will create the timeptr from the tms? If so, why not have this function do that?